hometask 5

MihailNenahov opened the pull request.

Commit:
finished 4 hometask

Commit:
corrected everythink except tests

Commit:
finished tests

Commit:
changed Less func

Commit:
removed fifth hw

Commit:
going to rewrite it another way

Commit:
5 - 1 task v 1.0

Commit:
finished rewriting and wrote tests(92.6% cover)

Commit:
finished 4 HW construct

Commit:
going to change my code for better cover

Commit:
initial commit

Commit:
убрал (if wrongData)

Commit:
rebuild coverout.html file

Commit:
tests reached 98,3%

Commit:
tests cover 100%

Commit:
finished debug and tests for sec task

Commit:
trying to rewrite programs to pass same test

Commit:
programs passed same test

Commit:
bench doesn't crash

Commit:
finished all tasks

burov commented on FifthTask/my_str_to_num/bench/myStrToNum1.go:
```diff
@@ -0,0 +1,305 @@
+	case "int":
+		{
+			return intConverter(givenString)
+		}
```
I don't think task asks to implement this

burov reviewed (changes requested):
I can't find a solution for any tasks which we provide, please remove redundant code and complete tasks according to their description

Commit:
rewrite the functions

Commit:
rewrited all code

Commit:
added bench conclusion

Commit:
removed redundant files

imatsiash-ib commented on FifthTask/my_str_to_num/solut/str_to_num.go:
```diff
@@ -0,0 +1,16 @@
+)
+
+func myStrToNum1(givenString string) (res int, err error) {
+	res, err = strconv.Atoi(givenString)
```
```suggestion
	return strconv.Atoi(givenString)
```

imatsiash-ib commented on FifthTask/my_str_to_num/solut/some_test.go:
```diff
@@ -0,0 +1,31 @@
+	receivedNumber, err = myStrToNum2(a)
+	fmt.Println(receivedNumber)
+	assert.NotNil(t, err, "test3, nil check")
+}
```
Please cover negative scenarios as well.

imatsiash-ib commented on FifthTask/my_str_to_num/solut/compare_test.go:
```diff
@@ -0,0 +1,18 @@
+
+func BenchmarkMyStrToNum1(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		_, _ = myStrToNum1(strconv.Itoa(i))
```
```suggestion
		myStrToNum1(strconv.Itoa(i))
```

imatsiash-ib commented on FifthTask/debug_some_code/someCodeToDebug.go:
```diff
@@ -0,0 +1,37 @@
+package main
```
Where you use these functions ?

imatsiash-ib replied:
Sorry, I didn't notice that this is a task3 solution.

MihailNenahov replied:
i don't know, what even this functions should do. if u'll left some comments. it would make me understand, what to do.

Commit:
after reading comments

added negative scenario
edited bench file
edited myStrToNum1
gave for dejson posisbility to show unmarshaled data
